### ISSUE TITLE
Running multiple server side tests in a single control file leads to multiple steps being repeated

### ISSUE THREAD

**Issue by aditijha** [autotest/autotest#987]
Hi,

I have a control file, control.srv, which runs several server side tests. It has a format similar to:

*
NAME = 'xyz'
AUTHOR = 'xyz'
TIME = 'MEDIUM'
TEST_CLASS = 'xyz'
TEST_CATEGORY = 'syz'
TEST_TYPE = 'server'
EXPERIMENTAL = 'False'

DOC = """
"""

def run(machine):
    host = hosts.create_host(machine)
    job.run_test('test1', host=host)
    job.run_test('test2', host=host)
    ...
    ...

job.parallel_simple(run, machines)
*

The tests run fine, however I noticed that for each run_test some steps are repeated (e.g. ssh to client, install autotest, etc) even though the tests are to run on the same client.
Is there another way to achieve a workflow like this, where all the setup steps run only once and and not during each test.

**Comment by lmr** [comment]
No. `parallel_simple` will spawn different subprocesses for each worker. What you can do is to make install autotest very quick and then it wouldn't be much of a problem.
